thirdparty/build-thirdparty.sh

```diff
@@ -1545,12 +1545,43 @@ build_concurrentqueue() {
     cp ./*.h "${TP_INSTALL_DIR}/include/"
 }
 
+#clucene
+build_clucene() {
+    if [[ -z ${USE_AVX2} ]]; then
+        USE_AVX2=1
+    fi
+    if [[ -z ${BUILD_TYPE} ]]; then
+        BUILD_TYPE=Release
+    fi
+    check_if_source_exist "${CLUCENE_SOURCE}"
+    cd "${TP_SOURCE_DIR}/${CLUCENE_SOURCE}"
+    mkdir -p "${BUILD_DIR}" && cd "${BUILD_DIR}"
+    rm -rf CMakeCache.txt CMakeFiles/
+
+    if [[ "${CC}" == *gcc ]]; then
+        CPPFLAGS="-fno-omit-frame-pointer -g -Wno-narrowing" \
+            CXXFLAGS="-fno-omit-frame-pointer -g -Wno-narrowing" \
+            ${CMAKE_CMD} -G "${GENERATOR}" -DCMAKE_INSTALL_PREFIX="${TP_INSTALL_DIR}" -DBUILD_STATIC_LIBRARIES=ON \
+            -DUSE_AVX2="${USE_AVX2}" -DCMAKE_BUILD_TYPE="${BUILD_TYPE}" -DBUILD_CONTRIBS_LIB=ON ..
+        ${BUILD_SYSTEM} -j "${PARALLEL}" && ${BUILD_SYSTEM} install
+    elif [[ "${CC}" == *clang ]]; then
+        CPPFLAGS="-fno-omit-frame-pointer -g -Wno-c++11-narrowing" \
+            CXXFLAGS="-fno-omit-frame-pointer -g -Wno-c++11-narrowing" \
+            ${CMAKE_CMD} -G "${GENERATOR}" -DCMAKE_INSTALL_PREFIX="${TP_INSTALL_DIR}" -DBUILD_STATIC_LIBRARIES=ON \
+            -DUSE_AVX2="${USE_AVX2}" -DCMAKE_BUILD_TYPE="${BUILD_TYPE}" -DBUILD_CONTRIBS_LIB=ON ..
```

Comment:
Looks verbose. The following common part can be separated like

```shell
${CMAKE_CMD} -G "${GENERATOR}" -DCMAKE_INSTALL_PREFIX="${TP_INSTALL_DIR}" -DBUILD_STATIC_LIBRARIES=ON \
            -DCMAKE_CXX_FLAGS="-fno-omit-frame-pointer ${warning_narrowing}"
            -DUSE_AVX2="${USE_AVX2}" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DBUILD_CONTRIBS_LIB=ON ..
${BUILD_SYSTEM} -j "${PARALLEL}"
${BUILD_SYSTEM} install
```

Reply:
done, good suggestion, thanks